plugins/inputs/logfile/fileconfig.go

```diff
@@ -43,7 +43,7 @@ type FileConfig struct {
 	//The regex of the timestampFromLogLine presents in the log entry
 	TimestampRegex string `toml:"timestamp_regex"`
 	//The timestampFromLogLine layout used in GoLang to parse the timestampFromLogLine.
-	TimestampLayout string `toml:"timestamp_layout"`
+	TimestampLayout []string `toml:"timestamp_layout"`
```

Comment:
@jefchien I thought you said not to do this

Reply:
We can't satisfy the requirement to default for `%m` and `%-m` without setting both options prior to getting the log timestamp. For `%d` and `%-d` we are able to achieve this by using `_2` but that's not an option for month. So this is the only way to get `timestamp_layout` to be compatible.

Reply:
@sethAmazon I thought we didn't need to do this, but since Go's time layout doesn't support zero padded/non-padded months with a single token, we have to have multiple layouts.